app/javascript/components/ImageModal.js

```diff
@@ -1,46 +1,55 @@
 import React from 'react'
-import { useSelector, useDispatch } from 'react-redux'
 import Modal from 'react-modal'
 import Uppy from '@uppy/core'
 import XHRUpload from '@uppy/xhr-upload'
 import { DragDrop } from '@uppy/react'
 
 function ImageModal() {
-  const dispatch = useDispatch()
-  const imagesState = useSelector((state) => state.images)
+  const [isAddingImage, setIsAddingImage] = React.useState(false)
+  React.useEffect(() => {
+    const handleEvent = () => {
+      setIsAddingImage(true)
+    }
+    window.addEventListener('ImageUploadRequested', handleEvent)
+    return () => window.removeEventListener('ImageUploadRequested', handleEvent)
+  }, [])
 
   const token = document.head.querySelector('[name~=csrf-token][content]')
     .content
 
-  var uppy = Uppy({
```

Comment:
memoized this so it doesn't create new Uppys on each render

Reply:
nice!